validator/gke_enable_shielded_nodes.rego

```diff
@@ -0,0 +1,67 @@
+#
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package templates.gcp.GCPGKEEnableShieldedNodesConstraintV1
+
+import data.validator.gcp.lib as lib
+
+deny[{
+	"msg": message,
+	"details": metadata,
+}] {
+	constraint := input.constraint
+	asset := input.asset
+	asset.asset_type == "container.googleapis.com/Cluster"
+
+	cluster := asset.resource.data
+	node_pools := lib.get_default(cluster, "nodePools", [])
+	node_pool := node_pools[_]
+
+	# We fail if either the cluster doesn't have shielded vms, or
+	# one of the node pools doesn't have the proper settings.
+	cluster_shielded_nodes(cluster, node_pool)
+
+	message := sprintf("Cluster %v doesn't use shielded VMs with integrity and secure boot in node pool %v.", [asset.name, node_pool.name])
+	metadata := {"resource": asset.name}
+}
+
+###########################
+# Rule Utilities
+###########################
+
+# Checks the Shielded Nodes setting in the cluster configuration.
+cluster_shielded_nodes(cluster, node_pool) {
```

Comment:
I think the naming on these functions should be clearer to show they are looking *for* violations (ie. return true when the cluster or node pool is *not* shielded.)

ex. `are_shielded_nodes_enabled`

Reply:
Changed all names to indicate that they are negatives.